recipes/configs/gemma2/2B_full.yaml

```diff
@@ -0,0 +1,74 @@
+# Config for multi-device full finetuning in full_finetune_distributed.py
+# using a gemma2 2B model
+#
+# This config assumes that you've run the following command before launching
+# this run:
+#   tune download google/gemma-2-2b --ignore-patterns "gemma-2-2b.gguf"  --hf-token <HF_TOKEN>
+#
+# To launch on 4 devices, run the following command from root:
+#   tune run --nnodes 1 --nproc_per_node 4 full_finetune_distributed --config gemma2/2B_full
```

Comment:
Maybe it's just me but when I try to run these distributed recipes I am hitting `AssertionError: FSDP requires named DeviceMesh dims for ND parallelism`. It looks to me like we are actually entering `_init_sharded_param` with a DTensor (see [here](https://github.com/pytorch/pytorch/blob/86b45bde196b976247f2771196f8c1c689dd66b7/torch/distributed/_composable/fsdp/_fsdp_param.py#L280)), which does not happen with our other recipes. Need to figure out why this would be happening

Reply:
Ah I think I cracked the case. See [here](https://github.com/pytorch/torchtune/pull/1835/files#r1817437522)

Reply:
Big mistake, thank you for catching that!

Reply:
The logs after this fix look much better than previously for the 9b single lora pipeline!
[log_gemma2-2b-single-lora_1729937021.txt](https://github.com/user-attachments/files/17533365/log_gemma2-2b-single-lora_1729937021.txt)

Reply:
@ebsmothers aren't the losses too low? could it be because of the (non) causal sliding window attention problem ?

Reply:
Sorry I missed this comment before. I am gonna run some of your configs on my end now so will get back to you